oxide-vpc/tests/integration_tests.rs

```diff
@@ -2024,3 +2046,369 @@ fn gateway_icmpv6_ping() {
+    g1.port.start();
+    set_state!(g1, PortState::Running);
+    let mut pcap = PcapBuilder::new("gateway_router_advert_reply.pcap");
+
+    // ====================================================
+    // Generate a Router Solicitation from G1 to Virtual GW
+    // ====================================================
+    let mut pkt1 = gen_router_solicitation(&g1_cfg.private_mac);
+    pcap.add_pkt(&pkt1);
+
+    // ================================================================
+    // Run the Solicitation through g1's port in the outbound
+    // direction and verify it results in an Router Advertisement
+    // hairpin back to guest.
+    // ================================================================
+    let res = g1.port.process(Out, &mut pkt1, &mut ameta);
+    let hp = match res {
+        Ok(Hairpin(hp)) => hp,
+        _ => panic!("expected Hairpin, got {:?}", res),
+    };
+    assert_port!(g1);
+
+    let reply = hp.parse().unwrap();
+    pcap.add_pkt(&reply);
+
+    // Ether + IPv6
+    assert_eq!(reply.body_offset(), ETHER_HDR_SZ + IPV6_HDR_SZ);
+    assert_eq!(reply.body_seg(), 0);
+
+    let meta = reply.meta();
+    assert!(meta.outer.ether.is_none());
+    assert!(meta.outer.ip.is_none());
+    assert!(meta.outer.ulp.is_none());
+
+    match meta.inner.ether.as_ref() {
+        Some(eth) => {
+            assert_eq!(
+                eth.src, g1_cfg.gateway_mac,
+                "Router advertisement should come from the gateway's MAC"
+            );
+            assert_eq!(
+                eth.dst, g1_cfg.private_mac,
+                "Router advertisement should be destined for the guest's MAC"
+            );
+        }
+
+        None => panic!("no inner ether header"),
+    }
+
+    let (src, dst) = match meta.inner.ip.as_ref().unwrap() {
+        IpMeta::Ip6(ip6) => {
+            assert_eq!(
+                ip6.src,
+                Ipv6Addr::from_eui64(&g1_cfg.gateway_mac),
+                "Router advertisement should come from the \
+                gateway's link-local IPv6 address, generated \
+                from the EUI-64 transform of its MAC",
+            );
+            let expected_dst = Ipv6Addr::from_eui64(&g1_cfg.private_mac);
+            assert_eq!(
+                ip6.dst, expected_dst,
+                "Router advertisement should be destined for \
+                the guest's Link-Local IPv6 address, generated from \
+                the EUI-64 transform of its MAC"
+            );
+            assert_eq!(ip6.proto, Protocol::ICMPv6);
+            (
+                Ipv6Address::from_bytes(ip6.src.bytes().as_slice()),
+                Ipv6Address::from_bytes(expected_dst.bytes().as_slice()),
+            )
+        }
+        ip4 => panic!("expected inner IPv6 metadata, got IPv4: {:?}", ip4),
+    };
+
+    let mut rdr = PacketReader::new(&reply, ());
+    // Need to seek to body.
+    rdr.seek(ETHER_HDR_SZ + IPV6_HDR_SZ).unwrap();
```

Comment:
You can replace this with `reply.get_body_rdr()`.

Reply:
Oh, nice didn't know about that. Thanks!